src/app/components/breadcrumb/breadcrumb.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, NavigationEnd, Router } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { Breadcrumb } from './breadcrumb';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BreadcrumbService {
+  private readonly _breadcrumbs$ = new BehaviorSubject<Breadcrumb[]>([]);
+  readonly breadcrumbs$ = this._breadcrumbs$.asObservable();
+
+  constructor(private router: Router) {
+    this.router.events
```

Comment:
I am not sure if this is needed, but maybe we should remove this code from the constructor and call it using a lifecycle hook (OnInit is probably the one we want to use).

Reply:
Don't believe that you can use lifecycle hooks in a service, but go ahead and refactor as you see fit.